Add Token Endpoint and add Authorization Discovery

dshanske opened the pull request:
This adds the discovery stuff from previous with the token endpoint items for today. It makes a bunch of changes recommended by @aaronpk as well.

Will likely hide the option to use an external token endpoint in a future version, as it seems that most people would want the built-in.

dshanske commented:
Because @aaronpk had an issue with using localhost in testing, I added the DEBUG file @pfefferle created for webmentions when he had the same problem. Really looking forward to getting this in.

pfefferle commented on includes/class-indieauth-authenticate.php:
```diff
@@ -81,50 +100,78 @@ public function verify_access_token( $token ) {
 		);
 		// redirect to authentication endpoint
-		wp_redirect( get_option( 'indieauth_authorization_endpoint' ) . '?' . $query );
+		wp_redirect( $authorization_endpoint . '?' . $query );
```
you should use a more solid method to add the query to the authorization endpoint. If hoster does not support `mod_rewrite`, `$authorization_endpoint` might already has query params.

pfefferle commented on includes/class-indieauth-token-endpoint.php:
```diff
@@ -0,0 +1,229 @@
+	}
+
+
+		/**
```
indents!

pfefferle commented on includes/class-indieauth-token-ui.php:
```diff
@@ -0,0 +1,112 @@
+	 * @access public
+	 */
+	public function options_form() {
+		echo '<div class="wrap">';
```
perhaps use a template, or at least use a more template style output (don't echo HTML tags)

pfefferle commented on functions.php:
```diff
@@ -0,0 +1,155 @@
+<?php
```
remove this file, because it is a duplicate of `includes/functions.php`

dshanske commented:
@pfefferle I removed the accidental adding of a duplicate functions file, changed the build_query to add_query_args which is more redundant, and moved the form to a template. Should do the trick.
